2023-Oct-18/Dave/direction.py

```diff
@@ -0,0 +1,29 @@
+from enum import Enum, EnumMeta
+from point import Point
+
+
+class DirectValueMeta(EnumMeta):
+    # This allows us to unpack enum members directly, in the format:
+    # row, col = Direction.RIGHT
+    def __getattribute__(cls, name):
+        value = super().__getattribute__(name)
+        if isinstance(value, cls):
+            value = value.value
+        return value
+
+    # This allows us to iterate through enum members in a for loop and access
+    # the Point value directly
+    def __iter__(cls):
+        for value in super().__iter__():
+            yield value.value
+
+
+class Direction(Enum, metaclass=DirectValueMeta):
```

Comment:
I'd be interested to hear a suggestion for an alternative. I definitely like the dot syntax. I had previously thought about using a named tuple, but something about that feels wrong, since there will only ever be one of these with fixed values and it seems odd to have the option to instantiate something that will only ever be one version of itself.

Open to suggestions, though.

Reply:
The C solution is macros (no in python thank god) The solution in other languages might be an enum or a hashmap. C++ has compile time constants (constexpr) so you can create a constexpr map or even a runtime map.